cpu/stm32_common/include/periph_cpu_common.h

```diff
@@ -150,6 +150,13 @@ typedef struct {
     uint8_t dma_stream;     /**< DMA stream used for TX */
     uint8_t dma_chan;       /**< DMA channel used for TX */
 #endif
+#ifdef UART_USE_HW_FC
+    uint8_t hw_flow_ctrl;   /**< use HW flow control flag */
```

Comment:
I would say we remove this flag, and test for hw flow control using the pin defs:
```c
if (uart_config[uart].cts_pin == GPIO_UNDEF) {
    // -> no flow control for this UART...
}
```